### PR TITLE
Refactor to work on dmd-2.027.2

### PR DESCRIPTION
**Pull request by ZeroSumGame**
The most recent D compiler changes broke the old way of referencing UDAs.  Some refactoring and code commenting was done to make it clearer what is going on in the code, and example usage added.

### CODE REVIEW

**Comment by b1naryth1ef** [comment]
This looks awesome. Can you make sure the comments are in the same style as prior (space before the comment body starts) and I can get this merged?

File: src/dscord/bot/command.d [hunk 1]
```diff
@@ -1,28 +1,80 @@
...
+//Note: The names *must* correspond to property names in CommandObject.
+enum TypeEnabled = "enabled";
+enum TypeTriggers = "triggers";
+enum TypeDescription = "description";
+enum TypeRegEx = "useRegex";
+enum TypeGroup = "group";
+enum TypeLevel = "level";
+
+//Template overloads for implementations of each type of CommandConfig info
+template CommandConfig(alias T) if(T==TypeTriggers){
```
Comment (b1naryth1ef) [inline src/dscord/bot/command.d:51]: for these I'd go with `if (T == TypeTriggers) {`, I tend to break std-D style a bit but I still prefer it.

File: src/dscord/bot/command.d [hunk 2]
```diff
@@ -1,28 +1,80 @@
...
+  static struct CommandConfig {
+    this(string[] args...){
+      triggers = args.dup;
+    }
+    string[] triggers;
+    //alias triggers this;
```
Comment (b1naryth1ef) [inline src/dscord/bot/command.d]: rm?

**Comment by ZeroSumGame** [comment]
Hope the updates have addressed stylistic issues.  Used an auto-formatter to normalize style since my editor normally uses three-space indents and it munched some spacing.

**Comment by b1naryth1ef** [comment]
Sorry for the wait on this, I'm trying to cleanup the JSON parsing before I merge anything else into master. Hoping to make some progress this weekend.

**Comment by b1naryth1ef** [comment]
This got merged in #17. Thanks again for putting this together!

**Pull request closed without merging**